course02/project01/gavin/Asteroids/collidable.cc

```diff
@@ -0,0 +1,21 @@
+#include <cmath>
+#include "collidable.h"
+#include "game_world.h"
+
+Collidable::Collidable(GameObject& parent, Tag tag, glm::vec2 position, float size) : GameObject(parent, tag, position, size)
+{
+    //FIXME: There has to be a better way
+    dynamic_cast<GameWorld*>(&mRoot)->AddCollider(this);
```

Comment:
This statement assumes that the root object is a 'GameWorld' and will throw an exception otherwise, I think it should be known at Compile time.

Reply:
Fixed in 76db252356e796ace0396de1cb5a68c9790e3d06. ```GameWorld``` is no longer a ```GameObject```.